File: nattybear/level1/문자열-다루기-기본/solution.js
function solution(s) {
  return four_or_six(s) && all_digit(s)
}

function four_or_six(s) {
  const size = s.length
  return size == 4 || size == 6
}

function digit(c) {
  return ('0' <= c && c <= '9')
}

function all_digit(s) {
  return s.split('').every(digit)
}

File: nattybear/level1/문자열-다루기-기본/Solution.java
class Solution {
  public boolean solution(String s) {
    return fourOrSix(s) && allDigit(s);
  }

  boolean fourOrSix(String s) {
    int size = s.length();
    return size == 4 || size == 6;
  }

  static boolean digit(int c) {
    return '0' <= c && c <= '9';
  }

  boolean allDigit(String s) {
    return s.chars().allMatch(Solution::digit);
  }
}
